samza-core/src/main/java/org/apache/samza/zk/ZkControllerImpl.java

```diff
@@ -87,6 +87,8 @@ public void stop() {
       zkLeaderElector.resignLeadership();
     }
 
+    zkUtils.deleteProcessorNode(processorId);
```

Comment:
Should this be done during finally so that it is always executed? it looks like any exception during `resignLeadership` will not trigger the delete?

Reply:
resignLeadership simply overrides a boolean variable's value.

Agree that it will be safer to move this to finally block. Moved it to finally block.

Reply:
`resignLeadership` is an interface - It's just that the Zk implementation is implementing it in a certain way by returning a boolean value.